.github/workflows/ci.yml

```diff
@@ -176,6 +176,107 @@ jobs:
               -expected-failure-list tests/expected-failure-github.txt \
               -expected-failure-list tests/expected-failure-record-replay-tests.txt
           fi
+      - name: Run Slang examples
```

Comment:
can we put this into a bash script file to simplify the yml file?

Reply:
Should be possible, I will try it out. I think I'll have to evaluate stuff like `"${{ matrix.os }}"` within the yaml file, but IMO it's cleaner to hoist that stuff out to the top level anyway, and just pass it as arguments to the script.

Reply:
Are there any examples that actually runs on macOS? If not I am OK with skipping macOS for now and just run the tests on Windows where we can use the most number of APIs.

Reply:
> Are there any examples that actually runs on macOS? If not I am OK with skipping macOS for now and just run the tests on Windows where we can use the most number of APIs.

I don't have a Mac to test on, but several of the examples seem to "pass" in CI.

In fact, even when I didn't run them headless with the `--test-mode` flag, several examples passed CI on MacOS.
(Initially I had implemented a `--single-frame-test-mode` flag that actually created a window and called `exit(0)` right after the first frame was presented. However that complicates CI because it would require a window system environment, so I just reused the existing headless rendering test mode instead.)

Reply:
Do you mean we should disable MacOS and/or Linux tests that are currently passing?